.github/workflows/qa.yaml

```diff
@@ -69,9 +74,23 @@ jobs:
           sudo -E $(which go) test -coverpkg=./... -coverprofile=/tmp/coverage.sudo.out -covermode=set $SUDO_PACKAGES
 
           # Combine coverage files, and filter out test utilities and generated files
-          cod_cov_dir="$(pwd)/coverage/codecov"
+          coverage_dir="$(pwd)/coverage"
+          cod_cov_dir="${coverage_dir}/codecov"
           mkdir -p "${cod_cov_dir}"
-          grep -hv -e "testutils" -e "pb.go:" -e "/e2e/" /tmp/coverage.out /tmp/coverage.sudo.out > "${cod_cov_dir}/coverage.combined.out"
+          combined_cov_file="${coverage_dir}/coverage.combined.out"
+          go_only_cov_file="${coverage_dir}/coverage.go-only.out"
+          echo "mode: set" > "${combined_cov_file}"
+          grep -hv -e "testutils" -e "pb.go:" -e "/e2e/" -e "mode: set" /tmp/coverage.out /tmp/coverage.sudo.out >> "${combined_cov_file}"
+
+          # Prepare XML coverage report
+          grep -hv -e "adsys-gpolist" -e "cert-autoenroll" "${combined_cov_file}" > "${go_only_cov_file}"
+          gocov convert "${go_only_cov_file}" | gocov-xml > "${coverage_dir}/coverage.xml"
+          reportgenerator -reports:"${coverage_dir}/*.xml" -targetdir:"${cod_cov_dir}" -reporttypes:Cobertura
```

Comment:
It’s more manual that I was initially hoping for, but I think this is what it is. We will notice when adding any non go file or files to ignore that the conversion will fail anyway.